### ISSUE TITLE
[Source Request]: 

### ISSUE THREAD

**Issue by garrathbond** [mampfes/hacs_waste_collection_schedule#2373]
### Municipality / Region

ipswich/uk

### Collection Calendar Webpage

?

### Example Address

?

### Collection Data Format

As ICS/ICAL file

### Additional Information

?

**Comment by garrathbond** [comment]
file:///C:/Users/gb1/Downloads/recycling-calendar.pdf

**Comment by garrathbond** [comment]
https://app.ipswich.gov.uk/bin-collection/pdf/702

**Comment by 5ila5** [comment]
Looks like I'm geo-blocked from Germany, and it won't load at all when opening through tor with GB exit node

**Comment by garrathbond** [comment]
not sure what that means

**Comment by 5ila5** [comment]
I cannot look into this as my requests are all blocked as I sit in Germany, and they seem to block all traffic that does not come from the UK

**Comment by garrathbond** [comment]
it is the uk! ipswich UK

**Comment by 5ila5** [comment]
Yes and I cannot look into it as I get this when visiting the bin-collection page from my GERMAN internet connection
![image](https://github.com/user-attachments/assets/c0a57ebb-0e1a-4662-9ac8-31174239a637)

**Comment by garrathbond** [comment]
well that seems very strange to me guess its a council thing (bit surprising)
[recycling-calendar (1).pdf](https://github.com/user-attachments/files/16714685/recycling-calendar.1.pdf)
does this help

**Comment by 5ila5** [comment]
It's not that uncommon to be GEO-blocked from some UK and other non EU sites. 

The PDF does not really help we do not parse PDFs we try to do the same web request the browser makes to get the dates shown on a webpage or to get an ICAL file (preferred but very rare for UK websites).

You can use the [static source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/source/static.md) and add your dates/intervals manually

**Comment by garrathbond** [comment]
ok thats a shame do you mean by static source it just a calander entry no read from any web site?

**Comment by 5ila5** [comment]
Yes, you can manually define dates and frequencies which will show up in your calendar without any web requests 

All details are here: https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/source/static.md

**Comment by CJEBond** [comment]
Ipswich has a new recycling service starting next month.

The new URL is:
https://app.ipswich.gov.uk/bin-collection-better-recycling/

On that page, you enter your street name. This links you to, for example:
https://app.ipswich.gov.uk/bin-collection-better-recycling/weeks/549
The next 4 collections are shown on that page, and you can download a PDF calendar.

I tried:
https://app.ipswich.gov.uk/bin-collection-better-recycling/months/549
This lists the dates for the next 6 months.

**Comment by markvp** [comment]
Thanks for the update @CJEBond — the new URL structure (`/bin-collection-better-recycling/months/{id}`) looks promising for a scraper-based source.

Unfortunately the site is geo-blocked outside the UK, so we can't investigate the API from here. To move this forward, we'd need someone based in the UK to:

1. Open browser devtools (Network tab) while searching for an address on the page
2. Identify what API request returns the zone/street ID (e.g. `549`) from a street name
3. Check what the `/months/{id}` page HTML looks like — specifically how dates and collection types are structured

If you're interested in implementing this yourself, the [South Ayrshire Council source](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/custom_components/waste_collection_schedule/waste_collection_schedule/source/south_ayrshire_gov_uk.py) is a good recent example of a two-step UK council source (address lookup → schedule fetch). Feel free to open a PR!

**Comment by CJEBond** [comment]
I've given it a go with some help from Claude. #6220
